### PR TITLE
Feature/eng 1280 Implement route output option

### PR DESCRIPTION
**Pull request by nilsbeck**
# Description
This PR adds the API for the `Output` option in router to achieve a customized fleet output.

## Changes
- `route/option.go` has now some more data structures

### CODE REVIEW

File: route/option.go [hunk 1]
```diff
@@ -328,6 +328,28 @@ func Sorter(
 	return sorterFunc(sorter)
 }
 
+// PlanMarshaller defines an interface that is used to override router's default
+// output when using multiple vehicles.
+type PlanMarshaller interface {
```
Comment (dirkschumacher) [inline route/option.go]: `Marshaller` -> `Marshaler`

File: route/option.go [hunk 2]
```diff
@@ -328,6 +328,28 @@ func Sorter(
...
+	Marshal(PartialPlan) ([]byte, error)
+}
+
+// VehicleMarshaller defines an interface that is used to override router's
+// default output when using exactly one vehicle.
+type VehicleMarshaller interface {
```
Comment (dirkschumacher) [inline route/option.go]: Can we call that `PartialVehicleMarshaler` to be inline with the interface

File: route/option.go [hunk 3]
```diff
@@ -328,6 +328,28 @@ func Sorter(
...
+	Marshal(PartialVehicle) ([]byte, error)
```
Comment (dirkschumacher) [inline route/option.go]: I would suggest we call the function `MarshalJSON` similar to the `Marshaler` interface in `encoding/json`.

**Review by dirkschumacher — COMMENTED** [review]
Just some quick comments even though still WIP

File: route/option.go (hunk 1 above)
Comment (nilsbeck) [reply route/option.go]: was removed in favor of other implementation

File: route/option.go (hunk 3 above)
Comment (nilsbeck) [reply route/option.go]: was removed in favor of other implementation

File: route/option.go (hunk 2 above)
Comment (nilsbeck) [reply route/option.go]: was removed in favor of other implementation

**Review by sebastian-quintero — DISMISSED** [review]

**Review by dirkschumacher — DISMISSED** [review]

**Review by sebastian-quintero — APPROVED** [review]

**Review by dirkschumacher — APPROVED** [review]

**Pull request merged**